lib/models/room.js

```diff
@@ -600,6 +604,30 @@ Room.prototype.addTags = function(event) {
     this.emit("Room.tags", event, this);
 };
 
+/**
+ * Update the account_data events for this room, overwriting events of the same type.
+ * @param {Array<MatrixEvent>} events an array of account_data events to add
+ */
+Room.prototype.addAccountData = function(events) {
+    for (var i = 0; i < events.length; i++) {
+        var event = events[i];
+        if (event.getType() === "m.tag") {
+            this.addTags(event);
+        }
```

Comment:
We should probably put a `continue` in this `if` block else `m.tag` events will be added to the `accountData` dict, which is presumably unintentional.